.github/workflows/reusable_provenance.yaml

```diff
@@ -114,44 +116,42 @@ jobs:
           # base64 --wrap=0 disables line wrapping, and encodes to base64 and outputs on a single line.
           echo "digest=$(sha256sum $(basename ${{ steps.artifact_path.outputs.artifact_path }}) | base64 --wrap=0)" >> $GITHUB_OUTPUT
 
-  # This job calls the generic SLSA workflow to generate provenance.
-  # See https://github.com/slsa-framework/slsa-github-generator/blob/main/internal/builders/generic/README.md
   generate_provenance:
-    if: github.event_name == 'push'
     needs: [build_binary]
     permissions:
       actions: read
       id-token: write
       contents: write
-    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v1.2.1
+    uses: ./.github/workflows/generate_provenance.yaml
```

Comment:
Will we get the correct information in the generated provenance this way? Since now technically the workflow is different. In other words: what do we need to check on the generated provenance to make sure it was genuinely created by the real SLSA generator?

Reply:
It is not very different from the previous situation. This only adds another nesting of reusable workflows. But I agree this is worth looking into.

Reply:
I check, in the newly generated provenances the entrypoint is the top-level workflow:

```
{
  "_type": "https://in-toto.io/Statement/v0.1",
  "predicateType": "https://slsa.dev/provenance/v0.2",
  "subject": [
    {
      "name": "oak_tensorflow_freestanding_bin",
      "digest": {
        "sha256": "7e96476a2a5f72e3070f79d37e97d99f127611768f816e8dd3e2be1491844107"
      }
    }
  ],
  "predicate": {
    "builder": {
      "id": "https://github.com/slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@refs/tags/v1.2.1"
    },
    "buildType": "https://github.com/slsa-framework/slsa-github-generator/generic@v1",
    "invocation": {
      "configSource": {
        "uri": "git+https://github.com/project-oak/oak@refs/heads/main",
        "digest": {
          "sha1": "2337f5c47b1c9642ab9d5222ede49ea9d9825f03"
        },
        "entryPoint": ".github/workflows/provenance.yaml"
      },
...
}
```

And the `builder.id` still is the slsa-generator workflow, and this `builder.id` is what we have to check.